src/test/java/teammates/test/cases/ui/StudentProfilePictureActionTest.java

```diff
@@ -189,22 +190,85 @@ private void testActionForStudentWithEmptyGoogleId() {
         assertEquals("", result.blobKey);
     }
 
-    protected void testActionWithEmailAndCourseUnauthorisedInstructor() {
+    protected void testActionWithEmailAndCourseUnauthorisedInstructorOrStudent() {
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.STUDENT_EMAIL, StringHelper.encrypt(student.email),
+                Const.ParamsNames.COURSE_ID, StringHelper.encrypt(student.course)
+        };
+        
         ______TS("Failure case: instructor not from same course");
         AccountAttributes unauthInstructor = dataBundle.accounts.get("instructor1OfCourse2");
         gaeSimulation.loginAsInstructor(unauthInstructor.googleId);
 
+        action = getAction(submissionParams);
+        try {
+            action.executeAndPostProcess();
+            signalFailureToDetectException("Unauthorised Access");
+        } catch (UnauthorizedAccessException uae) {
+            assertEquals("User is not in the course that student belongs to", uae.getMessage());
+        }
```

Comment:
@wkurniawan07 Sorry, I missed that. There was already an instructor with no "viewing student" privilege. I added test for that case.